### PR TITLE
Merge Login and Signup Epilogues

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref #13413, #13939

This is to merge the Epilogue feature branch into `develop`. It encompasses all changes for #13413 and #13939. 

---
Known Issues:
- Status bar text is white in light mode.
- Password field: when the view refreshes (i.e. changing split mode, toggling light/dark mode), the field is inset by a few points.
- Password field: Secure text icon color has not been updated.

---
To test:
- Login.
  - Verify the new epilogue appears as expected.
- Signup.
  - Verify the new epilogue appears as expected.

| Login | Signup |
|--------|-------|
| login | signup |

---
To show the Signup Epilogue without actually signing up:

- In WPiOS `Podfile`, point `WordPressAuthenticator` to your local Auth.
- In `LoginViewController`, change `showLoginEpilogue` to show the signup epilogue:

```
func showLoginEpilogue(for credentials: AuthenticatorCredentials) {
    guard let navigationController = navigationController else {
        fatalError()
    }
    
    showSignupEpilogue(for: credentials)
    
//        authenticationDelegate.presentLoginEpilogue(in: navigationController, for: credentials) { [weak self] in
//            self?.dismissBlock?(false)
//        }
}
```

- _Login_ with an existing account, and the _signup_ epilogue will be displayed.

---
PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
@bjtitus @Gio2018 - this PR is kind of big (since it's merging a feature branch). All the included changes have been reviewed on their respective PRs, but if either of you have time to help out @mindgraffiti and give this a look that'd be super awesome!

**Comment by ScoutHarris** [comment]
Thanks @mindgraffiti !

> The larger iPad (12.9", iOS 13.4.1) refused to draw border lines for a login epilogue that scrolled (multiple sites). A single site or a couple of sites drew in the border line just fine. 

🤔  I'm unable to reproduce this.

> The site icons show a light gray border initially drawn around them. If the site list scrolls, the table cells redraw and the border on the site icons disappear. 

I _finally_ saw this when I accidentally backgrounded the app. How you consistently see it I don't know. 😄 I'll take a look!

**Comment by ScoutHarris** [comment]
Hey @mindgraffiti .

> The site icons show a light gray border initially drawn around them.

Found it! It was a holdover from the old epilogue. And wasn't obvious if your sites have perfectly square icons. Like mine do. 😄 It's been removed. Thanks for repeatedly pointing it out until I paid attention. 🤦

**Comment by mindgraffiti** [comment]
Update: tested on my iPhone 11 Pro device (iOS 13.4.1). Did a quick smoke test of login flows with dark mode, light mode, No sites, 1 site, 3 sites, 100+ sites. Also smoke tested the sign up flow by email and uploaded a gravatar image / changed username option / set a weak password (prompt warned me to use a better password and refused to submit info), used a strong password. Once submitted, Create a Site epilogue appeared as expected. Once in, I added a self-hosted site to the account to check those flows.

File: WordPress/Classes/ViewRelated/NUX/LoginEpilogueBlogCell.swift [hunk 1]
```diff
@@ -22,4 +23,10 @@ class LoginEpilogueBlogCell: WPBlogTableViewCell {
             return siteIcon
         }
     }
+
+    func adjustSiteNameConstraint() {
+        // If the URL is nil, center the Site Name vertically with the site icon.
+        siteNameVerticalConstraint.isActive = (urlLabel?.text == nil)
+    }
+
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/NUX/LoginEpilogueBlogCell.swift:31]: These aren't "extra" spaces per se. I like to add lines between the last method in a class and the end of the class for readability. These are personal preferences and don't break any guidelines.

File: WordPress/Classes/ViewRelated/NUX/SignupEpilogueCell.swift [hunk 2]
```diff
@@ -8,7 +8,7 @@ protocol SignupEpilogueCellDelegate {
     func usernameSelected()
 }
 
-enum EpilogueCellType {
+enum EpilogueCellType: Int {
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/NUX/SignupEpilogueCell.swift:11]: 🤔 I don't see how this is better. `EpilogueCellType(rawValue:)` safely returns nothing if there is no matching value in the `enum`. 

However, while testing this I did notice `return super.tableView(tableView, cellForRowAt: indexPath)` crashes, so I'll fix that.

File: WordPress/Classes/ViewRelated/NUX/SignupEpilogueTableViewController.swift [hunk 3]
```diff
@@ -133,48 +136,33 @@ class SignupEpilogueTableViewController: NUXTableViewController, EpilogueUserInf
...
     }
 
     override func tableView(_ tableView: UITableView, estimatedHeightForFooterInSection section: Int) -> CGFloat {
         return Constants.headerFooterHeight
     }
 
     override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
-        if section == TableSections.password {
-            return UITableView.automaticDimension
+        guard section != TableSections.userInfo, showPassword else {
```
[the remark this answers is not in the document]
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/NUX/SignupEpilogueTableViewController.swift:161]: I thought about that actually. But with the compound conditions, it was less readable.

**Comment by ScoutHarris** [comment]
Thanks for your feedback @Gio2018 !

**Comment by designsimply** [comment]
Quick note to say that I removed the `[Status] Needs Design Review` label since I saw that @mattmiklic had reviewed the issue.

**Comment by ScoutHarris** [comment]
Hey thanks @designsimply . I totally missed that!

**Comment by mattmiklic** [comment]
Thanks for catching that @designsimply :)

**Pull request merged**